app/src/main/java/org/mozilla/reference/browser/settings/AccountSettingsFragment.kt

```diff
@@ -12,14 +13,66 @@ import androidx.preference.PreferenceFragmentCompat
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import mozilla.components.feature.sync.onSyncRunning
+import mozilla.components.feature.sync.onSyncIdle
 import org.mozilla.reference.browser.R
 import org.mozilla.reference.browser.ext.getPreferenceKey
 import org.mozilla.reference.browser.ext.requireComponents
 import org.mozilla.reference.browser.R.string.pref_key_sign_out
 import org.mozilla.reference.browser.R.string.pref_key_sync_now
 import org.mozilla.reference.browser.browser.FirefoxAccountsIntegration
+import org.mozilla.reference.browser.ext.components
+import java.lang.ref.WeakReference
+
+private fun WeakReference<AccountSettingsFragment>.ifAttached(
+    block: (fragment: AccountSettingsFragment, context: Context) -> Unit
+) {
+    this.get()?.let { fragment ->
+        fragment.context?.let { context ->
+            block(fragment, context)
+        }
+    }
+}
 
 class AccountSettingsFragment : PreferenceFragmentCompat() {
+    private val weakThis = WeakReference(this)
+
+    class OnSyncRunningCallback(private val fragmentRef: WeakReference<AccountSettingsFragment>) : onSyncRunning {
+        override fun invoke() {
+            fragmentRef.ifAttached { fragment, context ->
+                CoroutineScope(Dispatchers.Main).launch {
+                    val pref = fragment.findPreference(context.getPreferenceKey(pref_key_sync_now))
+                    pref.title = fragment.getString(R.string.syncing)
+                    pref.isEnabled = false
+                }
+            }
+        }
+    }
```

Comment:
See comment on AC PR. I think you can avoid a lot of that weak reference handling by using `ObserverRegistry`. Even if it does not apply you may want to make use of the lifecycle observer functionalities and not having to deal with weak references.